web/yo/app/scripts/directives/curationQueue.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Created by jiaojiao on 4/24/17.
+ */
+'use strict';
+
+/**
+ * @ngdoc directive
+ * @name oncokbApp.directive:curationQueue
+ * @description
+ * # curationQueue
+ */
+angular.module('oncokbApp')
+    .directive('curationQueue', function(DTColumnDefBuilder, DTOptionsBuilder, user, DatabaseConnector, $rootScope, $timeout) {
+        return {
+            templateUrl: 'views/curationQueue.html',
+            restrict: 'AE',
+            scope: {
+            },
+            replace: true,
+            link: {
+                pre: function preLink(scope) {
+                    scope.queue = [];
+                    scope.queueModel = $rootScope.model.getRoot().get('queue');
+                    _.each(scope.queueModel.asArray(), function(item) {
+                        scope.queue.push({article: item.get('article'), pmid: item.get('pmid'), link: item.get('link'), variant: item.get('variant'), addedBy: item.get('addedBy'), addedAt: item.get('addedAt'), curated: item.get('curated'), curator: item.get('curator')});
+                    });
+                    scope.dtOptions = {
+                        hasBootstrap: true,
+                        paging: false,
+                        scrollCollapse: true,
+                        aaSorting: [[0, 'asc']]
+                    };
+                    scope.dtColumns = [
+                        DTColumnDefBuilder.newColumnDef(0),
+                        DTColumnDefBuilder.newColumnDef(1),
+                        DTColumnDefBuilder.newColumnDef(2),
+                        DTColumnDefBuilder.newColumnDef(3).withOption('sType', 'date'),
+                        DTColumnDefBuilder.newColumnDef(4),
+                        DTColumnDefBuilder.newColumnDef(5).withOption('sType', 'curated'),
+                        DTColumnDefBuilder.newColumnDef(6)
+                    ];
+                },
+                post: function postLink(scope) {
+                    scope.$watch('article', function(n, o) {
+                        if(n && n !== o) {
+                            $timeout.cancel(scope.articleTimeoutPromise);
+                            scope.articleTimeoutPromise = $timeout(function() {
+                                if(/^[\d]*$/.test(scope.article)) {
+                                    scope.getArticle(scope.article);
+                                }
+                            }, 500);
+                        }
+                    });
+                }
+            },
+            controller: function($scope) {
+                $scope.addCuration = function() {
+                    var item = $rootScope.model.createMap();
```

Comment:
You can initial map in this fashion https://github.com/googledrive/realtime-playground/blob/master/elements/create-document.js#L93